src/Microsoft.AspNetCore.Sockets/ConnectionManager.cs

```diff
@@ -73,7 +74,11 @@ private static string MakeNewConnectionId()
 
         private static void Scan(object state)
         {
-            ((ConnectionManager)state).Scan();
+            // Don't trim timed out connections if the debugger is attached
+            if (!Debugger.IsAttached)
```

Comment:
I agree for now, but this is something that will mess with anyone trying to debug a SignalR Hub.